src/index.ts

```diff
@@ -0,0 +1,518 @@
+            const result = data[arg];
+            return typeof result === 'string' ? result :
+                   typeof result === 'number' ? `${result}` :
+                   exp;
+          });
+      }
+    }
+
+    // Fallback to the key.
+    return key;
+  }
+
+  protected callSubscribers(): void {
+    this.subscribers.map((subscriber) => subscriber());
+  }
+
+  /**
+   * Clones the translations object passed in. However, it will also verify that each context
+   * of a key is valid.
+   */
+  private cloneTranslations(translations: Translations): Translations {
+    const newTranslations = { ...translations };
+
+    Object
+      .keys(newTranslations)
+      .forEach((context) => {
+        if (process.env.NODE_ENV !== 'production') {
+          if (!context.match(VALID_KEY_CONTEXT)) {
+            throw new Error(
+              `${context} is not a valid key context. ` +
+              `Valid keys should be of the format ${VALID_KEY_CONTEXT.toString()}.`);
+          }
+        }
+
+        if (typeof newTranslations[context] === 'object') {
+          newTranslations[context] =
+            this.cloneTranslations(<Translations>newTranslations[context]);
+        }
+      });
+
+    return newTranslations;
+  }
+
+  private deleteFrom(locale: string | string[], lookup: Rules | Translations): void {
+    const locales = this.normalizeLocales(locale);
+    const mapper = (l: string) => {
+      if (l in lookup) {
+        delete lookup[l];
+        return true;
+      } else {
+        return false;
+      }
+    };
+
+    const hasChanges = locales.map(mapper).includes(true);
+
+    if (hasChanges) {
+      this.callSubscribers();
+    }
+  }
+
+  /**
+   * Looks up the appropriate key context for the locale and number.
+   */
+  private getPluralContext(locale: string, count: number): PluralCategory {
+    const { language } = new LanguageTag(locale);
+    const rule: Rule = this.rules[locale] || this.rules[language];
+
+    // Since the locale may be determined by the end user's environment, do not throw an error. For
+    // better debugging purposes, though, print a warning stating that there is no rule for this
+    // locale.
+    if (rule === undefined) {
+      if (this.pluralWarning === undefined) {
+        this.pluralWarning = new Set();
+      }
+
+      if (!this.pluralWarning.has(language)) {
+        this.pluralWarning.add(language);
+        console.warn(`No pluralization rule found for ${locale}.`);
+      }
+    }
+
+    return rule !== undefined ? rule(count) : other();
+  }
+
+  /**
+   * Generates all possible subsets from a language tag.
+   *
+   * Examples:
+   * ```
+   * new LanguageTag('zh-Hant-HK').makeSubsets()    // ['zh-Hant', 'zh']
+   * new LanguageTag('zh-HK').makeSubsets()         // ['zh']
+   * new LanguageTag('zh-Hant').makeSubsets()       // ['zh']
+   * new LanguageTag('zh').makeSubsets()            // []
+   * ```
+   */
+  private makeSubsets(locale: string): string[] {
```

Comment:
Docstring doesn't match implementation.

Reply:
Example?

Reply:
Docstring talks about a method called `makeSubsets` that takes no arguments and lives on `LanguageTag`. The method `makeSubsets` that follows the docstring takes a string argument and lives on `Monga`.